Lib/test/test_zipfile.py

```diff
@@ -2034,6 +2034,21 @@ def test_decompress_without_3rd_party_library(self):
             with zipfile.ZipFile(zip_file) as zf:
                 self.assertRaises(RuntimeError, zf.extract, 'a.txt')
 
+    def test_close_on_closed_file(self):
+        # BPO-41350: check whether ZipFile.close()
+        # incorrectly throws a ValueError exception
+        # when initialized with a file-like object
+        # that is closed before ZipFile.close() is
+        # called.
+        bytes_io = io.BytesIO()
+        zip_file = zipfile.ZipFile(bytes_io, mode="w")
+        bytes_io.close()
+
+        try:
+            zip_file.close()
+        except ValueError:
```

Comment:
This exception handling is unnecessary. Let the original ValueError fail the test.